pkg/ottl/contexts/internal/span.go

```diff
@@ -117,13 +123,13 @@ func SpanPathGetSetter[K SpanContext](path ottl.Path[K]) (ottl.GetSetter[K], err
 			case "message":
 				return accessStatusMessage[K](), nil
 			default:
-				return nil, fmt.Errorf("invalid span path expression %v", nextPath.Name())
+				return nil, FormatDefaultErrorMessage("status."+nextPath.Name(), SpanContextName, SpanRef)
 			}
-		} else {
-			return accessStatus[K](), nil
 		}
+		return accessStatus[K](), nil
+	default:
+		return nil, FormatDefaultErrorMessage(path.Name(), SpanContextName, SpanRef)
```

Comment:
Could we do some equivalent of a `String` method on path here that converts a path back to its string representation (i.e. all values of `Name` joined by `.`)? That would save us from having to handle each case above individually, and would help us here if a user entered something like `invalid.path.name`, which I think would just print out `invalid` right now.

Reply:
That was my first idea, but then `keys` made it tricker. If the entire path segment include keys it feels like they should be printed as well, but accessing a Key takes a `context.Context` and `K` because someday (soon<sup>TM</sup>) we'll allow dynamic indexing.

I think I could save off the original statement's text when creating the Path list.  I'll experiment with that.